portal-impl/src/com/liferay/portal/util/PortalInstances.java

```diff
@@ -261,13 +263,15 @@ public static long initCompany(Company company, boolean skipCheck) {
 		try {
 			CompanyThreadLocal.setCompanyId(company.getCompanyId());
 
-			if (!skipCheck) {
+			for (PortalFunction<String, Company> function : functions) {
```

Comment:
@slnn This one is pretty weird on its own, and when you get an array of the functions, you only execute the first one -- even more weird...

If you are going to move this part out of initCompany in the next PR, let's skip this change here.

Reply:
OK, got it

Reply:
That's the design, the array only has two status one is empty another is one item:  if we do not need check company, we pass nothing to the second parameters of initCompany(Company, PortalFunctions<String, Company>), otherwise, the array will only has one item.

Currently, based on my previous research, only one place needs to check the company, so I will skip the change here.